tests/ui/higher-ranked/leak-check/leak-check-in-selection-6-ambig-unify.rs

```diff
@@ -28,5 +29,5 @@ impl<V> Trait<u16, V> for () {}
 fn impls_trait<T: Trait<U, V>, U: Id<V>, V>() {}
 fn main() {
     impls_trait::<(), _, _>()
-    //~^ ERROR type annotations needed
+    //[old]~^ ERROR type annotations needed
```

Comment:
this test should be changed to have one step of indirection between the binder and the `LeakCheckFailure` bound.